### DISCUSSION TITLE
Duplicate product without image

### DISCUSSION THREAD

**Discussion by BenTen** [PrestaShop/PrestaShop#34016] [Ideas]
In Prestashop 1.5 (1.6?), it was possible to duplicate products without image.
Why this useful function has been remove from PS 1.7?

---

When you duplicate a product page with images, images are duplicate too.
In PS 1.5, it was possible to duplicate with or without image.

![Capture du 2019-04-15 15-57-55](https://user-images.githubusercontent.com/16067358/56143106-d7fea100-5f97-11e9-818c-4da388207cbd.png)

**Comment by khouloudbelguith** [comment]
Hi @BenTen,

It's possible to duplicate a product without an image with PrestaShop 1.7.
I attached a video that shows the duplication of a product without an image with version 1.7.5.1.
https://drive.google.com/file/d/1WoOVhECwyiGrcG-XkJBnm_WUu6tTTNR2/view

Thanks for check and feedback.

**Comment by BenTen** [comment]
Hello,
Thank you for your reply.
I mean, duplicate a product page with image.

When you duplicate a product page with images, images are duplicate too.
In PS 1.5, it was possible to duplicate with or without image.

**Comment by khouloudbelguith** [comment]
@BenTen, it's possible to duplicate a product with an image with PrestaShop 1.7.
I attached a video record.
https://drive.google.com/file/d/1xHd9Dta-l0_L8xCzsj5ARkrMnubYzmN5/view

Thanks for check and feedback.

**Comment by BenTen** [comment]
I mean:
Duplicate from product with images to product without image.

**Comment by khouloudbelguith** [comment]
@marionf, @colinegin, what do you think to add this option for version 1.7?
![Capture du 2019-04-15 15-57-55](https://user-images.githubusercontent.com/16067358/56143106-d7fea100-5f97-11e9-818c-4da388207cbd.png)

**Comment by colinegin** [comment]
Hello,

Thanks for your idea ! Could you give us more details about your need ? In which cases do you need this feature ?

**Comment by BenTen** [comment]
For example:
You create a product and you save it.

The next product that you need to create has the same features.
You will lost a lot of time to apply the same features if you remake from scratch the product page.

So now, you duplicate but if the product has a lot of pictures, you lost a lot of time to erase one by one the pictures.
 
So there is two things we can add:
**1/ A button "Duplicate without image" (like in PS 1.5) near of the other**
![7bd11524622a1b76b97970379f616b01.jpg](https://tof.cx/images/2019/04/25/7bd11524622a1b76b97970379f616b01.jpg)
and
[![f53a18c5bfdfe23c424379bc17f84bcd.md.jpg](https://tof.cx/images/2019/04/25/f53a18c5bfdfe23c424379bc17f84bcd.md.jpg)](https://tof.cx/image/EGd6c)

**2/ a button to delete all the images**
[![ea282deb5650560c248af84c311e0173.md.jpg](https://tof.cx/images/2019/04/25/ea282deb5650560c248af84c311e0173.md.jpg)](https://tof.cx/image/EG9He)

What do you think?

**Comment by ghost** [comment]
Hi @BenTen,

Thanks for your feedback 👍 
Great idea indeed!

@MatShir Do you think the first issue, adding the possibility to duplicate without images, should be added to the Product page migration rework?

Regarding the second issue "Delete all images", an issue has been opened specifically, see #12773

**Comment by MatShir** [comment]
Hi, 
Since it's an improvement I will see with the migration team if it is on the scope but meanwhile, I will propose your improvement in http://feedback.prestashop.com/forums/387864-prestashop-1-7-x

**Comment by kingjc666** [comment]
Come on guys, duplicate without images is aviable since 1.4 at least.
Its very annoying for stores with alot of similar products and different images.
Right now i have to duplicate product, delete the old images and add the new ones. Since i want to change only the images and the color to a duplicated product, i dont want to create it from scratch and fill in all the descriptions and metatitles again. Please make this happen!
Also when you duplicate a product it says Copy of a.. in the title. This is very annoying also.. the whole duplicate thing is useless right now.

**Comment by BenTen** [comment]
@kingjc666 For the "Copy of", you can modify \src\Adapter\Product\AdminProductDataUpdater.php

from
`	public function duplicateProduct($productId, $namePattern = 'copy of %s')`
to
`	public function duplicateProduct($productId, $namePattern = '%s')`

It's not clean because you need to modify core file but it works.

source: https://www.prestashop.com/forums/topic/925299-removing-copy-of-from-duplicated-products/

**Comment by kevdevfr** [comment]
> @kingjc666 For the "Copy of", you can modify \src\Adapter\Product\AdminProductDataUpdater.php
> 
> from
> ` public function duplicateProduct($productId, $namePattern = 'copy of %s')`
> to
> ` public function duplicateProduct($productId, $namePattern = '%s')`
> 
> It's not clean because you need to modify core file but it works.
> 
> source: https://www.prestashop.com/forums/topic/925299-removing-copy-of-from-duplicated-products/

Just wanna add, in the same core hacking fashion, it is possible to comment some lines in this file **\src\Adapter\Product\AdminProductDataUpdater.php**, in order to duplicate without images

```
            if (!Image::duplicateProductImages($id_product_old, $product->id, $combination_images)) {
                throw new UpdateProductException('An error occurred while copying images.', 5008);
            } else {
                $this->hookDispatcher->dispatchWithParameters('actionProductAdd', ['id_product_old' => $id_product_old, 'id_product' => (int) $product->id, 'product' => $product]);
                if (in_array($product->visibility, ['both', 'search']) && Configuration::get('PS_SEARCH_INDEXATION')) {
                    Search::indexation(false, $product->id);
                }

                return $product->id;
           }
```
would become
```
            /* if (!Image::duplicateProductImages($id_product_old, $product->id, $combination_images)) {
                throw new UpdateProductException('An error occurred while copying images.', 5008);
            } else { */
                $this->hookDispatcher->dispatchWithParameters('actionProductAdd', ['id_product_old' => $id_product_old, 'id_product' => (int) $product->id, 'product' => $product]);
                if (in_array($product->visibility, ['both', 'search']) && Configuration::get('PS_SEARCH_INDEXATION')) {
                    Search::indexation(false, $product->id);
                }

                return $product->id;
            /* } */
```

this way it doesn't call the _Image::duplicateProductImages_ and goes straight to finishing the new clone product.

**Comment by zuk3975** [comment]
there is still time to add this behavior to new product page here is the POC pr https://github.com/PrestaShop/PrestaShop/pull/25159
